ksql-execution/src/main/java/io/confluent/ksql/execution/plan/LogicalSchemaWithMetaAndKeyFields.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.plan;
+
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Objects;
+
+public final class LogicalSchemaWithMetaAndKeyFields {
```

Comment:
What's the benefit of this class vs just calling `schema.withAlias(alias).withMetaAndKeyFieldsInValue()` or `schema.withoutMetaAndKeyFieldsInValue().withoutAlias()` where is needed?

Reply:
It keeps the specifics of the transformation done by `StreamSource` within `StreamSource` and an explicit requirement of the interface. Without this, the contract would be that StreamSource expects a schema in its constructor that has had the alias and meta/key transformations already added, which is kind of brittle.